### PR TITLE
Avoid using inner classes in SpanData

### PR DESCRIPTION
**Pull request by bogdandrutu**
Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

### CODE REVIEW

File: sdk/trace/src/main/java/io/opentelemetry/sdk/trace/data/LinkData.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+/**
+ * Data representation of a Link.
+ *
+ * <p>Used (for example) in batching operations, where a single batch handler processes multiple
+ * requests from different traces. Link can be also used to reference spans from the same trace.
+ */
+public interface LinkData {
+
+  /**
+   * Returns a new immutable {@code Link}.
```
Comment (anuraaga) [inline sdk/trace/src/main/java/io/opentelemetry/sdk/trace/data/LinkData.java:21]:
```suggestion
   * Returns a new immutable {@link LinkData}.
```

If we change all these to `{@link}` than renames will be applied in the future (well hopefully we're GA and don't rename :)

File: sdk/trace/src/main/java/io/opentelemetry/sdk/trace/data/SpanData.java [hunk 2]
```diff
@@ -144,18 +141,19 @@ default String getParentSpanId() {
...
    *
    * @return The total number of events recorded on this span.
    */
   int getTotalRecordedEvents();
 
   /**
-   * The total number of {@link ImmutableLink} links that were recorded on this span. This number
-   * may be larger than the number of links that are attached to this span, if the total number
-   * recorded was greater than the configured maximum value. See: {@link
+   * The total number of {@link ImmutableLinkData} links that were recorded on this span. This
```
Comment (anuraaga) [inline sdk/trace/src/main/java/io/opentelemetry/sdk/trace/data/SpanData.java:154]:
```suggestion
   * The total number of {@link LinkData} links that were recorded on this span. This
```

**Review by anuraaga — APPROVED** [review]
Thanks!

File: exporters/jaeger/src/test/java/io/opentelemetry/exporter/jaeger/JaegerGrpcSpanExporterTest.java [hunk 3]
```diff
@@ -95,7 +95,7 @@ void testExport() throws Exception {
             .setName("GET /api/endpoint")
             .setStartEpochNanos(TimeUnit.MILLISECONDS.toNanos(startMs))
             .setEndEpochNanos(TimeUnit.MILLISECONDS.toNanos(endMs))
-            .setStatus(Status.ok())
+            .setStatus(StatusData.ok())
```
Comment (jkwatson) [inline exporters/jaeger/src/test/java/io/opentelemetry/exporter/jaeger/JaegerGrpcSpanExporterTest.java:98]: do we need a StatusData? Would the API's StatusCode work for this?

Comment (jkwatson) [reply exporters/jaeger/src/test/java/io/opentelemetry/exporter/jaeger/JaegerGrpcSpanExporterTest.java:98]: never mind. we need the description.

Comment (anuraaga) [reply exporters/jaeger/src/test/java/io/opentelemetry/exporter/jaeger/JaegerGrpcSpanExporterTest.java:98]: `StatusData` is `StatusCode` + (`String`) description

**Review by jkwatson — APPROVED** [review]

**Pull request merged**